packages/@uppy/status-bar/src/StatusBarUI.tsx

```diff
@@ -140,8 +140,7 @@ export default function StatusBarUI<M extends Meta, B extends Body>({
   const showUploadBtn =
     !error &&
     newFiles &&
-    !isUploadInProgress &&
-    !isAllPaused &&
+    ((!isUploadInProgress && !isAllPaused) || recoveredState) &&
```

Comment:
nit: it might be easier to follow the logic by rewriting it to:
```suggestion
    (recoveredState || !(isUploadInProgress || isAllPaused)) &&
```

Reply:
Somehow I read the first one quicker

Reply:
wdyt of

```js
  const showUploadBtn =
    !hideUploadButton &&
    !error &&
    newFiles &&
    allowNewUpload &&
    (
      recoveredState ||
      (!isUploadInProgress && !isAllPaused)
    )
```

Reply:
That's basically the same but different formatting? It's up to prettier to decide that.

Reply:
The order is different, and IMO easier to follow